libvast/src/system/partition.cpp

```diff
@@ -187,8 +187,12 @@ evaluate(const PartitionState& state, const expression& expr) {
 bool partition_selector::operator()(const vast::qualified_record_field& filter,
                                     const table_slice_column& x) const {
   auto& layout = x.slice->layout();
-  vast::qualified_record_field fqf{layout.name(), layout.fields.at(x.column)};
-  return filter == fqf;
+  if (filter.layout_name != layout.name())
```

Comment:
Can you keep the old code commented out here, saying that the below is essentially identical, and add another comment explaining what allocation you're avoiding here?